pkg/internal/telegram/types.go

```diff
@@ -0,0 +1,110 @@
+package telegram
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kubesphere/notification-manager/pkg/apis/v2beta2"
+	"github.com/kubesphere/notification-manager/pkg/constants"
+	"github.com/kubesphere/notification-manager/pkg/internal"
+	"github.com/modern-go/reflect2"
+)
+
+type Receiver struct {
+	*internal.Common
+	*Config
+
+	// The channel or user to send notifications to.
+	Channels       []string `json:"channels,omitempty"`
+	MentionedUsers []string `json:"mentionedUsers,omitempty"`
+}
+
+func NewReceiver(tenantID string, obj *v2beta2.Receiver) internal.Receiver {
+	if obj.Spec.Telegram == nil {
+		return nil
+	}
+	telegram := obj.Spec.Telegram
+	r := &Receiver{
+		Common: &internal.Common{
+			Name:           obj.Name,
+			TenantID:       tenantID,
+			Type:           constants.Telegram,
+			Labels:         obj.Labels,
+			Enable:         telegram.Enabled,
+			AlertSelector:  telegram.AlertSelector,
+			ConfigSelector: telegram.TelegramConfigSelector,
+			Template: internal.Template{
+				TmplName: *telegram.Template,
+				TmplText: telegram.TmplText,
+			},
+		},
+		Channels:       telegram.Channels,
+		MentionedUsers: telegram.MentionedUsers,
+	}
+
+	return r
```

Comment:
The receiver needs to set resourceversion too like [this](https://github.com/kubesphere/notification-manager/blob/master/pkg/internal/dingtalk/types.go#L55).